django/interface/views.py

```diff
@@ -16,8 +19,108 @@ def statistics(request):
 
 
 def estimate(request):
-    """View of interface estimate"""
-    return render(request, "interface/estimate.html")
+    """View of interface estimate
+    This function calculates the average stay of pets in the shelter based on the
+    get parameters.
+    If there are not enough pets to estimate the range is increased until there are
+    NUMBER_OF_PETS_NEEDED_FOR_ESTIMATE pets. Otherwise error is displayed.
+    If an estimate was calculated some example pets are displayed."""
+
+    context = {}
+
+    breeds = Pet.objects.values("breed").exclude(breed="").distinct()
+    context["breeds"] = breeds
+
+    filters = {
+        "breed": request.GET.get("breed"),
+        "age": request.GET.get("age"),
+        "weight": request.GET.get("weight"),
+        "gender": request.GET.get("sex"),
+    }
+
+    context.update(filters)
+
+    if filters["gender"] is not None:
+        filters["gender"] = filters["gender"].lower()
+
+    min_age = None
+    max_age = None
+    min_weight = None
+    max_weight = None
+
+    pets = Pet.objects.all()
+
+    if not (
+        (filters["breed"] is None)
+        and (filters["age"] is None)
+        and (filters["weight"] is None)
+        and (filters["gender"] is None)
+    ):
+        if filters["breed"] is not None and filters["breed"] != "":
+            pets = Pet.objects.filter(breed=filters["breed"])
+        if filters["gender"] is not None and filters["gender"] != "":
+            pets = pets.filter(gender=filters["gender"])
+        if filters["age"] is not None and filters["age"] != "":
+            min_age = int(filters["age"])
+            max_age = int(filters["age"])
+        if filters["weight"] is not None and filters["weight"] != "":
+            min_weight = int(filters["weight"])
+            max_weight = int(filters["weight"])
+
+    if pets.count() < NUMBER_OF_PETS_NEEDED_FOR_ESTIMATE:
+        context["error"] = "Not enough pets to estimate"
+        return render(request, "interface/estimate.html", context)
+
+    filtered_pets = filter_pets(pets, min_age, max_age, min_weight, max_weight)
+
+    sum_days = 0
+    pets_skipped = 0
+    for pet in filtered_pets:
+        if pet.date_in is None or pet.date_out is None:
+            pets_skipped += 1
+            continue
+        sum_days += (pet.date_out - pet.date_in).days
+
+    if filtered_pets.count() - pets_skipped < NUMBER_OF_PETS_NEEDED_FOR_ESTIMATE:
+        context["error"] = "Not enough pets to estimate"
+    else:
+        context["estimate"] = int(sum_days / (filtered_pets.count() - pets_skipped))
+        context["example_pets"] = filtered_pets.order_by("?")[:9]
+
+    return render(request, "interface/estimate.html", context)
```

Comment:
Is there a reason why age and weight can be modified both in the same tempo?
Age in month is usually greater than weight in kilograms

Reply:
The reason is just to increase the possible amount of data. Can you suggest a tempo?